crates/frontend/src/components/context_form/utils.rs

```diff
@@ -58,20 +58,25 @@ pub fn construct_context(
     conditions: Vec<(String, String, String)>,
     dimensions: Vec<Dimension>,
 ) -> Value {
-    let condition_schemas = conditions
-        .iter()
-        .map(|(variable, operator, value)| {
-            get_condition_schema(variable, operator, value, dimensions.clone()).unwrap()
-        })
-        .collect::<Vec<Value>>();
-
-    let context = if condition_schemas.len() == 1 {
-        condition_schemas[0].clone()
+    if conditions.is_empty() {
+        json!({})
     } else {
-        json!({ "and": condition_schemas })
-    };
+        let condition_schemas = conditions
+            .iter()
+            .map(|(variable, operator, value)| {
+                get_condition_schema(variable, operator, value, dimensions.clone())
+                    .unwrap()
```

Comment:
if this returns optional then use `unwrap_option_or_default_with_error`.

Reply:
Hey the change is the indentation here and also this returns Result<Value,String>